### PR TITLE
429 bytes without globalAlpha. It was gonna take the 🥾 sooner or later

### PR DESCRIPTION
**Pull request by p01**
Merge the loops
Use globalAlpha instead of fillStyle
Move density and force calculation to a function
Swap `wall` flag to be a `particle` flag

### CODE REVIEW

**Comment by xem** [comment]
Thanks! Do you think we should keep alpha filling? Maybe black is okay enough

**Comment by p01** [comment]
> Thanks! Do you think we should keep alpha filling? Maybe black is okay enough

I guess we were gonna get rid of the alpha sooner or later so let's go for it!

**429 bytes**

**Comment by p01** [comment]
Should we merge so others can improve upon this? @xem

File: index.html [hunk 1]
```diff
@@ -15,7 +15,7 @@ <h2>MiniFluid</h2>
 #  */p=a;              e,p&lt;r;                                        p+=5)//  #
 #  define              z(e,i)                                        f(p,p/*  #
 ## */[i]=e)f(q,w=cabs  (d=*p-  *q)/2-     1)if(0  &lt;(x=1-      w))p[i]+=w*/// ##
-   double complex a [  97687]  ,*p,*q     ,*r=a,  w=0,d;    int x,y;char b/* ##
+## double complex a [  97687]  ,*p,*q     ,*r=a,  w=0,d;    int x,y;char b/* ##
```
Comment (xem) [inline index.html:18]: this "  " was present in the original C file. I guess we can keep it until we do a JS quine ourselves :p

**Pull request merged**